rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -172,38 +172,35 @@ namespace rsx
 		void texture_read_semaphore_release(thread* rsx, u32 _reg, u32 arg)
 		{
 			// Pipeline barrier seems to be equivalent to a SHADER_READ stage barrier
-
-			const u32 index = method_registers.semaphore_offset_4097() >> 4;
 			// lle-gcm likes to inject system reserved semaphores, presumably for system/vsh usage
 			// Avoid calling render to avoid any havoc(flickering) they may cause from invalid flush/write
-
-			if (index > 63 && !rsx->do_method(NV4097_TEXTURE_READ_SEMAPHORE_RELEASE, arg))
+			const u32 offset = method_registers.semaphore_offset_4097() & -16u;
```

Comment:
If the only usable values of context are DMA_SEMAPHORE_R and DMA_SEMAPHORE_RW, isn't this new code the same as the old one?

Reply:
the register is allowed to point to DEVICE_RW, only the initial value after sys_rsx init point to semaphore context.

Reply:
is device_rw different from semaphore_rw? If so, doesn't this undo part of your 406e_semaphore address work?

Reply:
I think I've rushed out the tests to fix skate, didnt notice the result were incorrect as it seemed "fitting" with his skate's behavior. later attempts to reproduce this failed so im leaving device context as is for now.